extensions/amp-accordion/1.0/component.js

```diff
@@ -507,3 +507,29 @@ export function AccordionContent({
     </WithAmpContext>
   );
 }
+
+/**
+ * @param {{current: ?}|undefined} ref
+ * @return {Element|undefined}
+ */
+function checkRef(ref) {
+  if (!ref) {
+    return ref;
+  }
+
+  // in amp mode, the imperative handle gives us an object where the element
+  // property refers to the actual DOM element i.e. {element: DOMElement}
```

Comment:
Just for my understanding, why not just always return a DOM element? E.g. `useImperativeHandle(ref, () => contentElement, [contentElement])`. Would that completely clear up the delta between the two modes?

Reply:
@jridgewell Would you mind providing additional context?  I also do not have a good understanding, and made the change per my understanding of the requirements in the group chat for what's needed.  If there's a better way to do it such as:

```
ref.current = contentElement

or

useImperativeHandle(ref, () => contentElement, [contentElement])
```

I'd like to get a better understanding of what our options are and how we can best fit the needs for runtime.  Thanks!

Reply:
Is the `ref` passed to you from the outside? If so, there's one more nuance relevant here. A ref can be passed as a `Ref` object or as a function. E.g. someone might call a component like this:

```
<Component ref={(instance) => ...}></>
```

The `useImperativeHandle` takes care of that.

Reply:
> The useImperativeHandle takes care of that.

I'm trying to limit `useImperativeHandle` to expose a small API object. The current code just exports an element, without care for what is exposed on the object.

> Is the ref passed to you from the outside?

I believe it's internal only. It's always a ref object, never a function.

Reply:
If it's an internal use, we can certainly assume a specific form only, e.g. `ref.current`.